Platforms/SurfaceDuoFamilyPkg/Library/BootGraphicsProviderLib/BootGraphicsProviderLib.c

```diff
@@ -60,7 +59,14 @@ GetBootGraphic(
   //
   Status =
       GetSectionFromAnyFv(g, EFI_SECTION_RAW, 0, (VOID **)ImageData, ImageSize);
-
+  
+  // Identify Picture Format.
+  if ((*ImageSize > 4) &&
```

Comment:
Thanks tips I'll made modifications. 
By the way if buffer NULL the size should be less than 4? and if the first condition is not met，the rest conditions won't be checked because of  '&&'

Reply:
ImageSize is too a pointer, it can also be null, so you can't check it this way

Reply:
not to mention the pointer may point to a valid address anyhow (we have no os to safeguard random data) and we may be reading something completely wrong in ram here